packages/analytics-js/src/components/configManager/ConfigManager.ts

```diff
@@ -1,23 +1,168 @@
+import { filterEnabledDestination } from './util/filterDestinations';
 
-class ConfigManager {
+class ConfigManager implements IConfigManager {
+  httpClient: IHttpClient;
+  errorHandler?: IErrorHandler;
+  logger?: ILogger;
+  hasErrorHandler = false;
+  hasLogger = false;
+
+  constructor(httpClient: IHttpClient, errorHandler?: IErrorHandler, logger?: ILogger) {
+    this.errorHandler = errorHandler;
+    this.logger = logger;
+    this.httpClient = httpClient;
+    this.hasErrorHandler = Boolean(this.errorHandler);
+    this.hasLogger = Boolean(this.logger);
+  }
+
+  /**
+   * A function to validate, construct and store loadOption, lifecycle, source and destination
+   * config related information in global state
+   */
   init() {
-    // TODO: get writekey, dataplaneUrl and loadOptions from state to use
+    validateLoadArgs(state.lifecycle.writeKey.value, state.lifecycle.dataPlaneUrl.value);
+    const lockIntegrationsVersion = state.loadOptions.value.lockIntegrationsVersion === true;
+    // determine the path to fetch integration SDK url from
+    const intgCdnUrl = getIntegrationsCDNPath(
+      'process.package_version',
+      lockIntegrationsVersion,
+      state.loadOptions.value.destSDKBaseURL,
+    );
+
+    // determine if the staging SDK is being used
+    // TODO: deprecate this in new version and stop adding '-staging' in filenames
+    const { isStaging } = getSDKUrlInfo();
+
+    // set application lifecycle state in global state
+    batch(() => {
+      if (state.loadOptions.value.logLevel) {
+        state.lifecycle.logLevel.value = state.loadOptions.value.logLevel;
+      }
+      state.lifecycle.integrationsCDNPath.value = intgCdnUrl;
+      if (state.loadOptions.value.configUrl) {
+        state.lifecycle.sourceConfigUrl.value = `${state.loadOptions.value.configUrl}/sourceConfig/?p=process.module_type&v=process.package_version&writeKey=${state.lifecycle.writeKey.value}&lockIntegrationsVersion=${lockIntegrationsVersion}`;
+      }
+      state.lifecycle.isStaging.value = isStaging;
+    });
     this.getConfig();
   }
 
-  getConfig() {
-    // Once done call all ready
-    window.setTimeout(() => {
-      this.onReady();
-    }, 500);
+  /**
+   * A callback function that is executed once we fetch the source config response.
+   * Use to construct and store information that are dependent on the sourceConfig.
+   * @param res source config response
+   */
+  processConfig(response?: SourceConfigResponse | string) {
+    if (!response) {
+      throw Error('Unable to fetch source config');
+    }
+    let res: SourceConfigResponse;
+    const errMessage = 'Invalid source config';
+    try {
+      if (typeof response === 'string') {
+        res = JSON.parse(response);
+      } else {
+        res = response;
+      }
+    } catch (e) {
+      if (this.hasErrorHandler) {
+        this.errorHandler?.onError(e, 'Config Manager: processConfig');
+      }
+      throw Error(errMessage);
```

Comment:
Is this error handled at upper levels? 
If no, won't this be caught by the global unhandled error and notify again?
If yes, does it handle the error again?

Reply:
The error handler will take in the error, process it and if there is logger it will log in console the appropriate message and not throw again. https://github.com/rudderlabs/rudder-sdk-js/blob/v3-staging/packages/analytics-js/src/services/ErrorHandler/ErrorHandler.ts#L58

We need to sent any errors form our components to the error handler. only if one is not defined we should plainly throw.

please see here https://github.com/rudderlabs/rudder-sdk-js/blob/v3-staging/packages/analytics-js/src/services/HttpClient/HttpClient.ts#L72 how error handler is used (cerated the onError method and use in the rest of the code in this service)

Reply:
@MoumitaM, let's implement the same here.

Reply:
yes, lets be consistent on thin in our component classes (add an onError like the HttpClient one)

Reply:
Done